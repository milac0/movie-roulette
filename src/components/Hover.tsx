import React from "react";
import styles from "./hover.scss";

interface Props {
  onHover: React.ReactNode;
  children: React.ReactNode;
}

const Hover: React.FC<Props> = ({ onHover, children }) => {
  return (
    <div className={styles.hoverDiv}>
      <div className={styles.hoverCard}>
        <div className={styles.noHover}>{children}</div>
        <div className={styles.hover}>{onHover}</div>
      </div>
    </div>
  );
};

export default Hover;
