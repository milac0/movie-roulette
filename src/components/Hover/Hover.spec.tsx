import React from 'react';
import renderer from 'react-test-renderer';

import Hover from './Hover';

describe('Hover renders correctly', () => {
    test('snapshot renders', () => {
        const hover = <div>testing div!</div>
        const child = <div>another testing div!</div>

        const tree = renderer.create(
            <Hover onHover={hover}>
                {child}
            </Hover>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});

