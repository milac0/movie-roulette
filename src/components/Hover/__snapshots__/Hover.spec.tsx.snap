// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hover renders correctly snapshot renders 1`] = `
<div
  className="hoverCard"
>
  <div
    className="noHover"
  >
    <div>
      another testing div!
    </div>
  </div>
  <div
    className="hover"
  >
    <div>
      testing div!
    </div>
  </div>
</div>
`;
